### PR TITLE
test: Add conntrack entry timeout validation tests.

### LINKED ISSUE #9303: Improve unit tests for conntrack lookup in BPF
* [x] Merge initial framework PR #9304
* [x] Test the monitor notification behaviour with respect to new, established, closing, closed connection
  * [x] The first lookup should be monitored #9304
  * [x] Subsequent lookups within the timeout should not be monitored #9304
  * [x] After the monitor timeout has passed, a packet should be monitored #9304
  * [x] The first FIN in a direction should be monitored #9304
  * [x] Subsequent packets in the same direction do not need to be monitored #9771
  * [x] The second FIN (closing the other direction) should be monitored #9304
  * [x] Subsequent packets in the FIN direction do not need to be monitored #9304
* [x] Test the timeout behaviour of `__ct_lookup()`
  * [x] When only SYN was seen, there is the SYN timeout #9771
  * [x] When other subsequent packets were seen, there is the standard TCP lifetime timeout #9771
  * [x] Once bidirectional close (fin/rst) was seen, there is the FIN timeout #9771
* [x] When looking up a tuple that was not previously connection-tracked, the function should return CT_NEW. #10064

### PR DESCRIPTION
**Pull request by valas**
Partially fixes #9303.

Signed-off-by: Valas Valancius <valas@google.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9771)

### CODE REVIEW

**Comment by brb** [comment]
@valas Thanks for your contribution!

To make cilium-agent to compile, you need to run `make -C daemon apply-bindata`. See the following error from the Travis-CI build:
```
########################################################################

                  ERROR: bindata.go is out of date.

 This can happen for two reasons:

 1. You are using a go-bindata binary compiled with a different version

    of golang (not 1.13.5). If so, please up/downgrade.

 2. You have made changes to the bpf/ directory. Please run the

    following command to update the SHA in daemon/bpf.sha:

    $ make -C daemon apply-bindata

########################################################################
```

File: bpf/lib/conntrack_test.h [hunk 1]
```diff
@@ -90,41 +101,60 @@ static void test___ct_lookup()
 			  &ct_state, true, seen_flags, &monitor);
 	assert(res == CT_ESTABLISHED);
 	assert(monitor == TRACE_PAYLOAD_LEN);
+	assert(timeout_in(entry, CT_SYN_TIMEOUT));
 
 	/* Second with no new flags, no advance in time is not monitored */
+	advance_time();
```
Comment (joestringer) [inline bpf/lib/conntrack_test.h:108]: nit: The comment immediately above this line says 'no advance in time', is now inconsistent with this line.

**Review by joestringer — CHANGES_REQUESTED** [review]
Looks good, thanks for the submission! One minor nit below on an out-of-date comment line. I see that you also need to update the `bpf.sha` as @brb pointed out above.

File: bpf/lib/conntrack_test.h (hunk 1 above)
Comment (valas) [reply bpf/lib/conntrack_test.h:108]: Done. PTAL.

**Comment by valas** [comment]
Updated daemon/bpf.sha and tweaked the comments. PTAL.

**Comment by brb** [comment]
test-me-please

**Review by brb — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Pull request merged**